segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -614,6 +624,37 @@ public boolean isOffline() {
     }
 
     //endregion
+    public CompletableFuture<SnapshotInfo> readStorageSnapshot(Duration timeout) {
```

Comment:
Should we put these two methods in their own separate class? They only need the `ContainerMetadata` and a lambda reference to a method that looks like `updateAttributes`.

Separating it out would enable proper unit testing and not force us to make this method public (it has no reason to be).

Reply:
Made the methods private.